src/csharp/Grpc.IntegrationTesting/ServicesGrpc.cs

```diff
@@ -47,12 +47,28 @@ public interface IBenchmarkServiceClient
     }
 
     // server-side interface
+    [System.Obsolete("Service implementations should inherit from the generated abstract base class instead.")]
     public interface IBenchmarkService
     {
       Task<global::Grpc.Testing.SimpleResponse> UnaryCall(global::Grpc.Testing.SimpleRequest request, ServerCallContext context);
       Task StreamingCall(IAsyncStreamReader<global::Grpc.Testing.SimpleRequest> requestStream, IServerStreamWriter<global::Grpc.Testing.SimpleResponse> responseStream, ServerCallContext context);
     }
 
+    // server-side abstract class
+    public abstract class BenchmarkServiceBase
```

Comment:
Urgh - yes, that would be nasty. See https://blogs.msdn.microsoft.com/ericlippert/2010/03/09/do-not-name-a-class-the-same-as-its-namespace-part-one

Thinking about this further, for the *server*-side, this is probably okay - because we genuinely expect users to derive from it in order to implement the service. For the *client* side, we've already got the Client suffix, so we can presumably have an abstract HealthClient and an implementation of HealthClientImpl.